Assets/Scripts/Interactables/InteractableObject.cs

```diff
@@ -0,0 +1,36 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+[RequireComponent(typeof(Outline), typeof(SceneSwitcher))]
+public class InteractableObject : MonoBehaviour, IInteractable
+{
+    private Outline _outline;
+    private SceneSwitcher _sceneswitcher;
+    public string sceneName;
+
+    public void BeforeInteraction()
+    {
+        _outline.OutlineMode = Outline.Mode.OutlineVisible;
+    }
+
+    public void Interact()
+    {
+        _outline.OutlineMode = Outline.Mode.OutlineHidden;
+        StartCoroutine(_sceneswitcher.GoToNextScene(sceneName));
```

Comment:
Minor detail: You could use enums to determine existing scene names and to avoid errors